components/gateway/internal/auditlog/service.go

```diff
@@ -77,60 +64,107 @@ func NewService(client AuditlogClient, msgFactory AuditlogMessageFactory) *Servi
+			Name: "request",
+			Old:  "",
+			New:  msg.Request,
+		}, model.Attribute{
+			Name: "correlation-id",
+			Old:  "",
+			New:  correlationID,
+		})
+
+	err := svc.client.LogConfigurationChange(ctx, configChangeMsg)
+	return errors.Wrap(err, "while sending configuration pre-change")
+}
+
+func (svc *Service) Log(ctx context.Context, msg proxy.AuditlogMessage) error {
+	graphqlResponse, err := svc.parseResponse(msg.Response)
 	if err != nil {
 		return errors.Wrap(err, "while parsing response")
 	}
 
-	if len(graphqlResponse.Errors) == 0 {
-		log := svc.createConfigChangeMsg(claims, request)
-		log.Attributes = append(log.Attributes, model.Attribute{
-			Name: "response",
-			Old:  "",
-			New:  "success",
-		})
+	correlationID := msg.CorrelationIDHeaders[correlation.RequestIDHeaderKey]
 
-		err = svc.client.LogConfigurationChange(ctx, log)
+	if len(graphqlResponse.Errors) == 0 {
+		configChangeMsg := svc.createConfigChangeMsg(msg.Claims, msg.Request)
+		configChangeMsg.Attributes = append(configChangeMsg.Attributes,
+			model.Attribute{
+				Name: "request",
+				Old:  "",
+				New:  msg.Request,
+			}, model.Attribute{
+				Name: "response",
+				Old:  "",
+				New:  "success",
+			}, model.Attribute{
+				Name: "correlation-id",
+				Old:  "",
+				New:  correlationID,
+			})
+
+		err = svc.client.LogConfigurationChange(ctx, configChangeMsg)
 		return errors.Wrap(err, "while sending to auditlog")
 	}
 
 	if svc.hasInsufficientScopeError(graphqlResponse.Errors) {
-		msg := svc.msgFactory.CreateSecurityEvent()
-		eventData := model.SecurityEventData{ID: fillID(claims, "Security Event"), Reason: graphqlResponse.Errors}
+		securityEventMsg := svc.msgFactory.CreateSecurityEvent()
+		eventData := model.SecurityEventData{ID: fillID(msg.Claims, "Security Event"), Reason: graphqlResponse.Errors}
 		data, err := json.Marshal(&eventData)
 		if err != nil {
 			return errors.Wrap(err, "while marshalling security event data")
 		}
 
-		msg.Data = string(data)
-		err = svc.client.LogSecurityEvent(ctx, msg)
+		securityEventMsg.Data = string(data)
+		err = svc.client.LogSecurityEvent(ctx, securityEventMsg)
 		return errors.Wrap(err, "while sending security event to auditlog")
 	}
 
-	isReadErr, err := isReadError(graphqlResponse, request)
+	isReadErr, err := isReadError(graphqlResponse, msg.Request)
 	if err != nil {
 		return errors.Wrap(err, "while checking if error is read error")
 	}
 
-	msg := svc.createConfigChangeMsg(claims, request)
+	configChangeMsg := svc.createConfigChangeMsg(msg.Claims, msg.Request)
 	if isReadErr {
-		msg.Attributes = append(msg.Attributes, model.Attribute{
-			Name: "response",
-			Old:  "",
-			New:  "success",
-		})
-		err := svc.client.LogConfigurationChange(ctx, msg)
+		configChangeMsg.Attributes = append(configChangeMsg.Attributes,
+			model.Attribute{
+				Name: "request",
+				Old:  "",
+				New:  msg.Request,
+			}, model.Attribute{
+				Name: "response",
+				Old:  "",
+				New:  "success",
+			}, model.Attribute{
```

Comment:
Same as above correlation id. Check all places where correlation id is added as attribute and move it to the constructor